include/exadg/aero_acoustic/calculators/source_term_calculator.h

```diff
@@ -108,16 +116,30 @@ class SourceTermCalculator
           dpdt.submit_value(-rho * dpdt.get_value(q), q);
       }
 
+      // apply blend in function before integration
+      if(data.blend_in && data.blend_in_function->needs_evaluation_at_time(time))
```

Comment:
1. `&&` -> `and`

2. shared_ptr `data.blend_in_function` might be invalid, so this line could be problematic in my opinion. What is your intention? Should a user always have to define a blend_in_function, or only when `data.blend_in = true`? I think one should write two consecutive if-statements. The assert on `data.blend_in_function.get()` could be done outside the cell loop, or even during setup.

Reply:
The order of the comparisons is crucial: If `data.blend_in==false`, the second condition will not be evaluated. I can see how this is not optimal during reading the code and am happy to change it once we agree on the further procedure in the other comments.

Reply:
Personally, I would like that we do not expect ExaDG developers to have this detailed knowledge about C++. Also C++ books like "Einstieg in C++" von Arnold Willemer (4. Auflage, S. 91/92) clearly do not recommend to realize some non-trivial logic inside if statements.

Reply:
@nfehn I agree that non-trivial logic should be avoided inside `if` statements, but I have a pretty strong opinion that checking if a pointer is valid (first condition) and if yes, evaluate the data structure behind the pointer (second condition) is among the self-explanatory conditions of the C++ programming language. In my opinion, having two nested `if` statements is more difficult to read.

I would however suggest to write the check in a different form if possible,
```suggestion
      if(data.blend_in_function.get() != nullptr && data.blend_in_function->needs_evaluation_at_time(time))
```
because this more clearly documents the intent of the two checks.